app/services/preserved_object_handler.rb

```diff
@@ -60,7 +60,66 @@ def create
   end
 
   def check_existence
-    # this is just a stub
+    if invalid?
+      handler_results.add_result(PreservedObjectHandlerResults::INVALID_ARGUMENTS, errors.full_messages)
+    elsif PreservedObject.exists?(druid: druid)
+      Rails.logger.debug "check_existence #{druid} called"
+      if endpoint.endpoint_type.endpoint_class == 'online'
+        transaction_ok = with_active_record_transaction_and_rescue do
+          pres_object = PreservedObject.find_by!(druid: druid)
+          # FIXME: what if there is more than one associated pres_copy?
+          pres_copy = PreservedCopy.find_by!(preserved_object: pres_object, endpoint: endpoint) if pres_object
+
+          if pres_copy.version != pres_object.current_version
+            handler_results.add_result(PreservedObjectHandlerResults::PC_PO_VERSION_MISMATCH, pres_copy.class.name)
+            raise ActiveRecord::Rollback, 'PO current_version != PC version'
+          end
+
+          if incoming_version == pres_copy.version
+            handler_results.add_result(PreservedObjectHandlerResults::VERSION_MATCHES, pres_copy.class.name)
+            handler_results.add_result(PreservedObjectHandlerResults::VERSION_MATCHES, pres_object.class.name)
+            update_pc_validation_timestamps(pres_copy)
+            update_db_object(pres_copy)
+          else
+            version_comp_result = if incoming_version > pres_copy.version
+                                    PreservedObjectHandlerResults::ARG_VERSION_GREATER_THAN_DB_OBJECT
+                                  else
+                                    PreservedObjectHandlerResults::ARG_VERSION_LESS_THAN_DB_OBJECT
+                                  end
```

Comment:
unfortunately, I believe this is incorrect implementation.

I believe we need to proceed to do the changes to the catalog below when incoming > catalog, but we are to report the error ("loudly") if the incoming < catalog -- we are definitely NOT to update the version.

I suspect the specs written by Sarav and I were changed in a mistaken understanding of the expected behavior here.  

In the interests of forward progress, I have created a new ticket about this: #424.

Reply:
d'oh.  yeah, i think we got this case wrong, in that we always update version.  good catch.